### PR TITLE
[GPU] Update TopK to support non const top_k input

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Details:
 - *Update TopK to support parameter input*
 - *Refactor primitive_impl create for arg_max_min*
 - *Add new constructors for new shape infer*

### Tickets:
 - *98114*

### CODE REVIEW

**Review by e-ddykim — APPROVED** [review]
Looks good to me

File: src/plugins/intel_gpu/include/intel_gpu/primitives/arg_max_min.hpp [hunk 1]
```diff
@@ -43,6 +43,42 @@ struct arg_max_min : public primitive_base<arg_max_min> {
...
+                const size_t num_outputs = 1)
+        : primitive_base(id, {input}, {output_padding}, {optional_data_type{output_data_type}}, num_outputs),
+          mode(mode),
+          top_k(top_k),
+          axis(axis),
+          sort(sort),
+          values_first(values_first) {}
+
+    /// @brief Constructs arg_max_min for top_k parameter
+    arg_max_min(const primitive_id& id,
```
Comment (yeonbok) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/arg_max_min.hpp:65]: BTW can't we use this as the first API?

Comment (yeonbok) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/arg_max_min.hpp:65]: I think we can set top_k value in creation as 0 if the 2nd node is not constant.

**Comment by andrew-k-park** [comment]
Overall looks good to me, could you resolve conflicts?

File: src/tests/functional/plugin/gpu/single_layer_tests/dynamic/top_k.cpp [hunk 2]
```diff
@@ -19,16 +19,16 @@ using namespace ov::test;
...
+        int64_t,                            // keepK
+        int64_t,                            // axis
+        ngraph::opset4::TopK::Mode,         // mode
+        ngraph::opset4::TopK::SortType,     // sort
+        ElementType,                        // Net precision
+        ElementType,                        // Input precision
+        ElementType,                        // Output precision
+        InputShape,                         // inputShape
+        TargetDevice,                       // Device name
+        bool                                // Input style (true:constant, false:parameter)
```
Comment (andrew-k-park) [inline src/tests/functional/plugin/gpu/single_layer_tests/dynamic/top_k.cpp:31]: nit: you can use `ngraph::helpers::InputLayerType` to distinguish layer type for code readability

File: src/plugins/intel_gpu/include/intel_gpu/primitives/arg_max_min.hpp (hunk 1 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/arg_max_min.hpp]: done

File: src/tests/functional/plugin/gpu/single_layer_tests/dynamic/top_k.cpp (hunk 2 above)
Comment (kelvinchoi-intel) [reply src/tests/functional/plugin/gpu/single_layer_tests/dynamic/top_k.cpp]: done

**Pull request merged**